packages/components/navigation/BuyTokens.tsx

```diff
@@ -1,32 +1,33 @@
 import React from "react";
-import { Linking, TextStyle, TouchableOpacity } from "react-native";
+import { TextStyle } from "react-native";
 import { CreditCardIcon } from "react-native-heroicons/outline";
 
 import { primaryColor } from "../../utils/style/colors";
 import { layout } from "../../utils/style/layout";
 import { BrandText } from "../BrandText";
+import { OmniLink } from "../OmniLink";
 
 export const BuyTokens: React.FC<{
   flexDirection: "row" | "column";
   textStyle: TextStyle;
 }> = ({ flexDirection, textStyle }) => {
   return (
-    <TouchableOpacity
+    <OmniLink
+      to={{
+        screen: "Swap",
+      }}
       style={{
         justifyContent: "center",
         alignItems: "center",
         marginBottom: layout.spacing_x1,
         flexDirection,
       }}
-      onPress={() => {
-        Linking.openURL("https://frontier.osmosis.zone/?from=OSMO&to=TORI");
-      }}
     >
       <CreditCardIcon
         color={primaryColor}
         style={{ marginRight: flexDirection === "row" ? layout.spacing_x1 : 0 }}
       />
-      <BrandText style={textStyle}>Buy Tokens</BrandText>
-    </TouchableOpacity>
+      <BrandText style={textStyle}>Buy TORI</BrandText>
```

Comment:
is this alway TORI ? can we get token name from network.currencies[0].displayName or sth like that ?

Reply:
Yes, always tori.